### PR TITLE
[in-review] Code optimisations via adding the credo

### PR DESCRIPTION
**Pull request by imahmedismail**
This PR involves the following changes:

- Add credo to the project and:
   - Run mix credo --all to identify possible code optimizations
   - Resolve most of the errors generated by credo such as:
     - Numbers larger than 9999 should be written with underscores: 58_127
     - Modules should have a @moduledoc tag
     - Comparison will always return true

### CODE REVIEW

File: test/blanks_test.exs [hunk 1]
```diff
@@ -19,7 +19,7 @@ defmodule BlanksTest do
   end
 
   test "multiple arguments" do
-    ast = quote do: assert(___ == ___)
+    ast = quote do: assert true == false
```
Comment (iamvery) [inline test/blanks_test.exs:22]:
```suggestion
    ast = quote do: assert ___ == ___
```

**Review by iamvery — CHANGES_REQUESTED** [review]
This all seems pretty reasonable to me, and I don't have any strong feelings against Credo. There is one place that I think is a mistake (see requested change).

File: test/blanks_test.exs (hunk 1 above)
Comment (imahmedismail) [reply test/blanks_test.exs]: This is done.

**Comment by iamvery** [comment]
I added configuration for credo to run in CI using `--strict` mode and fixed the suggestions it made. Thanks for the idea and getting this started!

**Pull request merged**